web/src/hashchange.js

```diff
@@ -471,6 +471,8 @@ function hashchanged(from_reload, e) {
         return undefined;
     }
 
+    popovers.hide_all();
```

Comment:
Original Comment on `web/src/user_card_popover.js` line 367:
> sahil: I guess this is for hiding the popover when pressing browser back button. I think we should update the code in hashchange.js. You can see we have hide_all calls, which are used to hide all popovers, etc. there.

I've deleted the new event listeners for closing the popovers and instead refactored `hashchange.js` to achieve the same effect.